base/src/com/google/idea/blaze/base/sync/ProjectStateSyncTask.java

```diff
@@ -189,6 +182,29 @@ private SyncProjectState getProjectState(BlazeContext context, BlazeSyncParams p
         .build();
   }
 
+  private ListenableFuture<BlazeInfo> createBazelInfoFuture(
+          BlazeContext context,
+          List<String> syncFlags,
+          SyncMode syncMode) {
+    boolean useBazelInfoRunner = !BlazeInfoProvider.isEnabled() || syncMode == SyncMode.FULL;
```

Comment:
Maybe we could just pass the sync mode to the `BlazeInfoProvider` and have it invalidate itself (or pretend that it has invalidated) on a full sync? I quite like the idea of "everything goes through the provider".

Reply:
The idea was to keep `BlazeInfoProvider` useful to other use cases apart from the sync, but if we are not planning to extend it, we can make it more coupled to the sync and pass the sync mode as well.